dd-java-agent/instrumentation/synapse-3.0/src/main/java/datadog/trace/instrumentation/synapse3/SynapseServerWorkerInstrumentation.java

```diff
@@ -66,26 +68,32 @@ public static void createWorker(@Advice.Argument(0) final SourceRequest request)
 
   public static final class ServerWorkerResponseAdvice {
     @Advice.OnMethodEnter(suppress = Throwable.class)
-    public static AgentScope beginResponse(
+    public static ContextScope beginResponse(
         @Advice.FieldValue("request") final SourceRequest request) {
       AgentScope.Continuation continuation =
           (AgentScope.Continuation)
               request.getConnection().getContext().removeAttribute(SYNAPSE_CONTINUATION_KEY);
       if (null != continuation) {
-        return continuation.activate();
+        // Activate the continuation to get the span, then attach it to current context
+        AgentScope agentScope = continuation.activate();
+        try {
+          return agentScope.span().attach();
```

Comment:
```suggestion
          return Context.current().with(agentScope.span()).attach();
```
Should this be like the above?